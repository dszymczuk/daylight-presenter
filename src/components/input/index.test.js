import React from "react";
import { render, cleanup } from '@testing-library/react';

import Input from "./index";

afterEach(cleanup);

it("renders Input", () => {
  const props = {
    label: "Sample label",
    field: {
      name: "sampleName",
      value: "",
      onChange: () => {
      }
    }, form: {
      errors: []
    }
  };

  const { asFragment } = render(<Input {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders Input with errors", () => {
  const props = {
    label: "Sample label",
    field: {
      name: "sampleName",
      value: "",
      onChange: () => {
      }
    }, form: {
      errors: {
        sampleName: "Sample error"
      }
    }
  };

  const { asFragment } = render(<Input {...props} />);
  expect(asFragment()).toMatchSnapshot();
});