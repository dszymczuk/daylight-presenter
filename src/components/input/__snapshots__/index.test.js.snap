// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="sampleName"
    >
      Sample label
    </label>
    <input
      class="form-control"
      id="sampleName"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`renders Input with errors 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="sampleName"
    >
      Sample label
    </label>
    <input
      class="form-control is-invalid"
      id="sampleName"
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    >
      Sample error
    </div>
  </div>
</DocumentFragment>
`;
